zrcs/zrc-2.md

```diff
@@ -0,0 +1,326 @@
+| --------- | ------------- | -------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| eventName | `MoveSuccess` | Sending is successful.     | `from`: `ByStr20` which is the sender's address, `to`: `ByStr20` which is the recipient's address, and `amount`: `Uint128` which is the amount to the sent. |
+| eventName | `Error`       | Sending is not successful. | - emit `CodeNotFound` if the balance is not found. <br> - emit `CodeNotAuthorized` if the balance is insufficient.                                          |
+
+#### 2. OperatorSend
+
+```ocaml
+(* @dev: Moves amount tokens from token_owner to recipient. _sender must be an operator of token_owner. *)
+(* @param from:        Address of the token_owner whose balance is decreased.                           *)
+(* @param to:          Address of the recipient whose balance is increased.                             *)
+(* @param amount:      Amount of tokens to be sent.                                                     *)
+transition OperatorSend(from: ByStr20, to: ByStr20, amount: Uint128)
+```
+
+|        | Name     | Type      | Description                                              |
+| ------ | -------- | --------- | -------------------------------------------------------- |
+| @param | `from`   | `ByStr20` | Address of the token_owner whose balance is to decrease. |
+| @param | `to`     | `ByStr20` | Address of the recipient whose balance is to increase.   |
+| @param | `amount` | `Uint128` | Amount of tokens to be sent.                             |
+
+|           | Name          | Description                | Event Parameters                                                                                                                                            |
+| --------- | ------------- | -------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| eventName | `MoveSuccess` | Sending is successful.     | `from`: `ByStr20` which is the sender's address, `to`: `ByStr20` which is the recipient's address, and `amount`: `Uint128` which is the amount to the sent. |
+| eventName | `Error`       | Sending is not successful. | - emit `CodeNotFound` if the balance is not found. <br> - emit `CodeNotAuthorized` if operator is not approved **or** if the balance is insufficient.       |
+
+#### 3. TransferFrom
+
+```ocaml
+(* @dev: Move a given amount of tokens from one address to another using the allowance mechanism. The caller must be an approved_spender. *)
+(* @param spender: Address of the spender who is tied to the token_owner's assets and have spending allowances allocated by token_owner.  *)
```

Comment:
Similar to the fix done in `OperatorSend`, the `spender` parameter shouldn't be explicit, but should be assumed and validated to be `_sender`.